frontend/src/actions/authActions.ts

```diff
@@ -112,15 +113,22 @@ export const userLogOut = () => (dispatch: Dispatch<logoutUserAction | CallHisto
     }
 }
 
+
+
 export const recover = (payload: IUserEmail) =>
   async (dispatch:Dispatch<logoutUserAction | CallHistoryMethodAction>): Promise<void> => {
     const authApiClient = new AuthApiClient()
 
     try {
-      const result = await authApiClient.recover(payload)
-      const {message} = result
-      toast.error(message)
-      dispatch(push('/login'))
+      const {email} = payload
+      if (emailFormatIsValid(email)) {
+        const result = await authApiClient.recover(payload)
+        const {message} = result
+        toast.info(message)
+        dispatch(push('/login'))
+      } else {
+        toast.error('Invalid e-mail format.')
```

Comment:
As a user I would prefer that mistakes I made in form will be shown to me near exact form field where they were made, not as toast on top of screen

Reply:
Ok, done